java/src/main/java/org/wildfly/openssl/OpenSSlSession.java

```diff
@@ -197,11 +197,11 @@ public Principal getLocalPrincipal() {
         if (local == null || local.length == 0) {
             return null;
         }
-        return principal(local);
+        return firstCertificate(local).getIssuerX500Principal();
```

Comment:
Shouldn't this one also be changed to ```getSubjectX500Principal()``` since the [JDK implementation](http://hg.openjdk.java.net/jdk8u/jdk8u/jdk/file/d5a00b1e8f78/src/share/classes/sun/security/ssl/SSLSessionImpl.java#l568) for ```getLocalPrincipal()``` also uses that?

Reply:
Done